api/orders.js

```diff
@@ -281,8 +281,223 @@ function cancelOrder(request, response, next) {
+ */
+function getOrderBook(request, response, next) {
+  var options = request.params;
+
+  Object.keys(request.query).forEach(function(param) {
+    options[param] = request.query[param];
+  });
+
+  parseOptions(options)
+  .then(validateOptions)
+  .then(getLastValidatedLedger)
+  .then(getBidsAndAsks)
+  .spread(respondWithOrderBook)
+  .catch(next);
+
+  function parseOptions(options) {
+    var baseCurrencyArray    = options.base.split('+');
+    var counterCurrencyArray = options.counter.split('+');
+
+    options.validated = true;
+    options.order_book = options.base + '/' + options.counter;
+
+    options.base = {
+      currency: (baseCurrencyArray.length >= 1 ? baseCurrencyArray[0] : ''),
+      issuer:   (baseCurrencyArray.length >= 2 ? baseCurrencyArray[1] : '')
+    };
+
+    options.counter = {
+      currency: (counterCurrencyArray.length >= 1 ? counterCurrencyArray[0] : ''),
+      issuer:   (counterCurrencyArray.length >= 2 ? counterCurrencyArray[1] : '')
+    };
+
+    return Promise.resolve(options);
+  }
+
+  function validateOptions(options) {
+    return new Promise(function(resolve, reject) {
+      if (!ripple.UInt160.is_valid(options.account)) {
+        reject(new InvalidRequestError('Parameter is not a valid Ripple address: account'));
+      }
+
+      if (!options.base.currency) {
+        reject(new InvalidRequestError('Invalid parameter: base. Must be a currency string in the form currency+counterparty'));
+      }
+
+      if (!validator.isValid(options.base.currency, 'Currency')) {
+        reject(new InvalidRequestError('Invalid parameter: base. Must be a currency string in the form currency+counterparty'));
+      }
+
+      if (options.base.currency !== 'XRP' && (!options.base.issuer || !ripple.UInt160.is_valid(options.base.issuer))) {
+        reject(new InvalidRequestError('Invalid parameter: base. Must be a currency string in the form currency+counterparty'));
+      }
+
+      if (!options.counter.currency) {
+        reject(new InvalidRequestError('Invalid parameter: counter. Must be a currency string in the form currency+counterparty'));
+      }
+
+      if (!validator.isValid(options.counter.currency, 'Currency')) {
+        reject(new InvalidRequestError('Invalid parameter: counter. Must be a currency string in the form currency+counterparty'));
+      }
+
+      if (options.counter.currency !== 'XRP' && (!options.counter.issuer || !ripple.UInt160.is_valid(options.counter.issuer))) {
+        reject(new InvalidRequestError('Invalid parameter: counter. Must be a currency string in the form currency+counterparty'));
+      }
+
+      if (options.counter.currency === 'XRP' && options.counter.issuer) {
+        reject(new InvalidRequestError('Invalid parameter: counter. XRP cannot have counterparty'));
+      }
+
+      if (options.base.currency === 'XRP' && options.base.issuer) {
+        reject(new InvalidRequestError('Invalid parameter: base. XRP cannot have counterparty'));
+      }
```

Comment:
I was thinking of adding something to `validator` that is something along the lines of `isValidXRPAmount` which will just contain the check `amount.currency === 'XRP' && amount.issuer`. Or.... alternatively, `utils.parseCurrencyQuery` could throw if an XRP currency was provided with issuer